manim/mobject/text/code_mobject.py

```diff
@@ -81,557 +59,207 @@ def construct(self):
-        possible options, visit https://pygments.org/docs/lexers/ and look for
-        'aliases or short names'.
-    generate_html_file
-        Defines whether to generate highlighted html code to the folder `assets/codes/generated_html_files`. Defaults to `False`.
-    warn_missing_font
-        If True (default), Manim will issue a warning if the font does not exist in the
-        (case-sensitive) list of fonts returned from `manimpango.list_fonts()`.
-
-    Attributes
-    ----------
-    background_mobject : :class:`~.VGroup`
-        The background of the code listing.
-    line_numbers : :class:`~.Paragraph`
-        The line numbers for the code listing. Empty, if
-        ``insert_line_no=False`` has been specified.
-    code : :class:`~.Paragraph`
-        The highlighted code.
-
+        The color of the border of the background. Defaults to ``WHITE``.
+    background_fill_color
+        The color of the background. Defaults to ``ManimColor("#222")``.
+    background_corner_radius
+        The corner radius of the background. Defaults to 0.2.
+    font
+        The font to use for the code. Defaults to ``"Monospace"``.
+    font_size
+        The font size to use for the code. Defaults to 24.
+    line_spacing
+        The amount of space between lines in relation to the font size.
+        Defaults to 0.5.
+    **paragraph_kwargs
+        Additional keyword arguments passed to :class:`.Paragraph`.
     """
 
-    # tuples in the form (name, aliases, filetypes, mimetypes)
-    # 'language' is aliases or short names
-    # For more information about pygments.lexers visit https://pygments.org/docs/lexers/
-    # from pygments.lexers import get_all_lexers
-    # all_lexers = get_all_lexers()
     _styles_list_cache: list[str] | None = None
-    # For more information about pygments.styles visit https://pygments.org/docs/styles/
 
     def __init__(
         self,
-        file_name: str | os.PathLike | None = None,
-        code: str | None = None,
-        tab_width: int = 3,
-        line_spacing: float = 0.3,
-        font_size: float = 24,
-        font: str = "Monospace",  # This should be in the font list on all platforms.
-        stroke_width: float = 0,
-        margin: float = 0.3,
-        indentation_chars: str = "    ",
-        background: str = "rectangle",  # or window
-        background_stroke_width: float = 1,
-        background_stroke_color: str = WHITE,
-        corner_radius: float = 0.2,
-        insert_line_no: bool = True,
-        line_no_from: int = 1,
-        line_no_buff: float = 0.4,
-        style: str = "vim",
+        code_file: StrPath | None = None,
+        code_string: str | None = None,
         language: str | None = None,
-        generate_html_file: bool = False,
-        warn_missing_font: bool = True,
-        **kwargs,
+        formatter_style: str = "vim",
+        tab_width: int = 4,
+        line_numbers: bool = True,
```

Comment:
The name should reflect that this is, indeed, a boolean. Maybe something like `add_line_numbers` or `use_line_numbers`?
```suggestion
        add_line_numbers: bool = True,
```

Reply:
Agreed! Went with `add_line_numbers` -- f3692312